src/simulations/optimisation.py

```diff
@@ -91,7 +91,7 @@ def solve(self, prm):
                 eta_old = copy.deepcopy(prm['car']['etach'])
                 print(f"prm['grd']['loads'][0][0][0] = "
                       f"{prm['grd']['loads'][0][0][0]}")
-                res = self._problem(prm)
+                res, prm = self._problem(prm)
```

Comment:
the parameters have already been added as an attribute of the `Optimiser` class in the `_update_prm` method do you do not need to input `prm` to the _problem method. see next comment